### PR TITLE
ddev: init at 1.21.4

### PR DESCRIPTION
**Pull request by jgonyea**
###### Description of changes

This PR is for adding a new package: ddev 
"Docker-based local PHP + Node.js web development environments"

Homepage: https://ddev.readthedocs.io/en/stable/
Source: https://github.com/drud/ddev

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drupol** [comment]
Hello,

Since ddev rely on Docker, I believe you should add it as a dependency, or else it won't work at all.

**Comment by jgonyea** [comment]
@drupol , that makes sense.  I added both docker and docker-compose.

**Comment by jgonyea** [comment]
@drupol , is there anything else I should do on this PR?

Thanks in advance.

**Comment by drupol** [comment]
Hi,

I don't know since I never used `ddev`.

Leave it open, someone else will most probably give some feedback at some point.

**Review by JulienMalka — CHANGES_REQUESTED** [review]
Could you squash your commits into 2 commits:
- maintainers: add jgonyea
- ddev: init at 1.21.4

Thanks!

File: pkgs/applications/virtualization/ddev/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+, ddev }:
+
+buildGoPackage rec {
+  pname = "ddev";
+  version = "1.21.4";
+  goPackagePath = "github.com/drud/ddev";
+
+  src = fetchFromGitHub {
+    owner = "drud";
+    repo = "ddev";
```
Comment (JulienMalka) [inline pkgs/applications/virtualization/ddev/default.nix:16]:
```suggestion
    repo = pname;
```

**Comment by jgonyea** [comment]
> Could you squash your commits into 2 commits:

@JulienMalka 
I commingled the new author and initial ddev project in my first commit.  Should I roll it all back and submit more cleanly?

**Comment by JulienMalka** [comment]
> > Could you squash your commits into 2 commits:
> 
> @JulienMalka I commingled the new author and initial ddev project in my first commit. Should I roll it all back and submit more cleanly?

You probably want to do something like ``git reset HEAD~4``, then commit again in the right order and by splitting the commits like explain, then push again using ``git push --force``

**Comment by jgonyea** [comment]
@JulienMalka, thanks.  I force pushed those requested changes.  Please let me know if there's anything else I can do.

**Review by JulienMalka — APPROVED** [review]
lgtm

**Comment by jgonyea** [comment]
Anything more that I should do on this for merging?  This is my first submission, so I don't know the process.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/virtualization/ddev/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+stdenv.mkDerivation {
+  name = pname;
+  version = version;
+  src = fetchFromGitHub {
+    owner = "ddev";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Wjg0Yxo/ulY6R6hhUMFvNZUSwpXENmAHU7GPbgdw7tw=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/ddev/default.nix:28]:
```suggestion
stdenv.mkDerivation rec {
  pname = "ddev";
  version = "1.21.6";

  src = fetchFromGitHub {
    owner = "ddev";
    repo = pname;
    rev = "v${version}";
    sha256 = "sha256-Wjg0Yxo/ulY6R6hhUMFvNZUSwpXENmAHU7GPbgdw7tw=";
  };

```

**Comment by SuperSandro2000** [comment]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

File: pkgs/applications/virtualization/ddev/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  buildInputs = [ bash curl git go ];
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/ddev/default.nix:29]: Please patch that the build requires a git clone. We can replace the values returned by git with dummy values or version.

```
build flags: SHELL=/nix/store/zlf0f88vj30sc7567b80l52d19pbdmy2-bash-5.2-p15/bin/bash build
fatal: not a git repository (or any of the parent directories): .git
make: *** [Makefile:96: .gotmp/bin/darwin_arm64/mkcert] Error 6
``` 

Also please split the packages for cross compilation into nativeBuildInputs and buildInputs.

**Comment by jgonyea** [comment]
Sorry, I didn't mean to leave this PR open.  I need to rework this.  Attempting to use the upstream Makefile fails due to the Makefile downloading external files (binary for mkcert).

**Comment by rfay** [comment]
@jgonyea I'm happy to help you get this going again.

**Comment by gbytedev** [comment]
@rfay I created a buildable ddev package and [put it on the NUR](https://nur.nix-community.org/repos/gbytedev/) for the time being. I spoke to @jgonyea and the plan is for me to [create a new RP from that soon](https://gbyte.dev/blog/use-ddev-nix-and-nixos).

**Comment by rfay** [comment]
Nice! Is there a way to get notifications for that? How does it get updated (since we're at v1.22.1?)

**Comment by gbytedev** [comment]
> Nice! Is there a way to get notifications for that? How does it get updated (since we're at v1.22.1?)

Use the configuration from the [blog post](https://gbyte.dev/blog/use-ddev-nix-and-nixos) and the package will update along with the system. In terms of it being up to date in the NUR, it seems I am responsible for now. I subscribe to ddev and will be pushing relatively timely updates.

I believe you can subscribe to [my repository](https://github.com/gbytedev/nur-packages) directly to get updates on the package - this is where the NUR runner is pulling changes from.

**Comment by drupol** [comment]
Sorry for the interruption but... why is `ddev` not in `nixpkgs` ?

**Comment by gbytedev** [comment]
> Sorry for the interruption but... why is `ddev` not in `nixpkgs` ?

I mentioned earlier that I intend to create a PR very soon. Still testing things ATM.

**Comment by rfay** [comment]
Subscribed, thanks.

**Comment by gbytedev** [comment]
> Subscribed, thanks.

One more thing, keep in mind the NUR runner is active once a day or so so you won't see changes right after I push an update to my repo. :/

**Comment by gbytedev** [comment]
@star-szr Feel free add me; I'm happy to deprecate my NUR package and support the upstream one as it has just been merged apparently. 🙌

**Comment by jgonyea** [comment]
I have nothing to add, except for thanks for all the work on getting this package working. I lacked the skills necessary to make this a proper release.  I'll be glad to help test once it goes live

**Pull request closed without merging**